src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1183,12 +1270,104 @@ case class Select(posn: Position, lhs: AST, rhs: String) extends AST(posn, lhs)
   }
 }
 
+case class ArrayDeclaration(posn: Position, elements: Array[AST]) extends AST(posn, elements) {
```

Comment:
You declare a variable or a function (generally, something with a name) but you create, construct or instantiate an object.